iio/iio_app/iio_app.c

```diff
@@ -241,7 +241,11 @@ static int32_t uart_setup(struct no_os_uart_desc **uart_desc,
 		/* TODO: remove this ifdef when asynchrounous rx is implemented on every platform. */
 #if defined(STM32_PLATFORM) || defined(MAXIM_PLATFORM) || defined(ADUCM_PLATFORM) || defined(PICO_PLATFORM)
 		.irq_id = uart_init_par->irq_id,
+#ifdef NO_OS_IIOD_UART_ASYNC_RX_FALSE
```

Comment:
I think it would be cleaner to just pass `uart_init_par->asynchronous_rx` directly, since that field is set by the user.